rust/core-lib/src/client.rs

```diff
@@ -235,6 +235,17 @@ impl Client {
     pub fn schedule_timer(&self, timeout: Instant, token: usize) {
         self.0.schedule_timer(timeout, token);
     }
+
+    /// Let client know tail was successfully enabled/disabled
```

Comment:
What are the reasons for a client to disable this?

Reply:
If client doesnt want to tail the file anymore, it'll disable tailing.

Reply:
If client doesnt want to tail file anymore.

Reply:
Ah, but wouldn't disabling tailing imply that the whole file is reloaded when something changes then (since inotify is still enabled)? I don't see the advantage of disabling tailing

Reply:
That is good point. So if i open up a log file and later on it gets constantly updated, should Xi also update the contents? Without explicitly giving the option to enable/disable, the end user experience would not be great as the file is constantly changing even though it might not be desirable. Also tailing mimics the `tail` command. Just like in in `tail` when you are at the bottom of file it'll keep scrolling, when you slightly scroll up, it stops. This behavior needs to be controllable too I think.
Having said that, I still think the way file is loaded currently is not efficient. Instead of loading delta, we load the whole file. I think this is what you are talking about. I havent changed existing things.

Reply:
> I think this is what you are talking about. I havent changed existing things.

Ah no, I not exactly and it's entirely fine that you're not touching that. I was just curious: If you disabling tailing, won't file reloading be enabled instead and as such the editor will keep loading new contents added to the file? The only differences I see between having tailing enabled/disabled:

* With tailing enabled you'll scroll if you're at the end of the file
* Reloading the entire file is less efficient

So wouldn't just having a switch to disable the scrolling do the trick? You'd have efficient adding of content without scrolling. But maybe I just misunderstand something :)

Reply:
Ok I see your point now. Thing is when tailing a file, I **APPEND** delta to the file. When tailing is disabled, all bets are off. A change in file (when tailing is disabled) might or might not be a delta change at the end of file. In that case it is harder to update the file I think. And for this reason, I think, we load the entire file when tailing is off.

Reply:
Ah, alrighty. I had expected that if I have a file:

```
line 1
line 2
line 3
line 4
```

and I add a new line between line 2 and 3, line 2.5 that we'd only reload the file from line 2 then, but I  suppose that was wrong and I don't know inotify too well :)

Sounds good to me then.

Reply:
Yeah. It gets a lot more complicated here. Besides, notify will only tell us something changed. It doesnt tell where what or where the change occurred.